airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -149,12 +159,15 @@ public void run(final ConnectionUpdaterInput connectionUpdaterInput) throws Retr
 
   private CancellationScope generateSyncWorkflowRunnable(final ConnectionUpdaterInput connectionUpdaterInput) {
     return Workflow.newCancellationScope(() -> {
-      connectionId = connectionUpdaterInput.getConnectionId();
+      if (workflowState.isDeleted()) {
+        log.info("Returning from workflow cancellation scope because workflow deletion was requested.");
+        return;
+      }
 
-      // Clean the job state by failing any jobs for this connection that are currently non-terminal.
-      // This catches cases where the temporal workflow was terminated and restarted while a job was
-      // actively running, leaving that job in an orphaned and non-terminal state.
-      ensureCleanJobState(connectionUpdaterInput);
+      if (workflowState.isUpdated()) {
+        // Act as a return
+        prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
+      }
```

Comment:
Since these conditions are exiting this scope, I kind of felt it was strange to have them in the middle of configuration operations like they were previously so I moved them up.

Also, is there a reason these have to be in the `cancellationScope`? Can we move this into the general `run` function? The `isDeleted` check seems to happening twice here, once on Line 134 and once in the sync workflow, why is that needed?

Reply:
It seems like we could just move the internals of the `isDeleted` check on line 134 into the `isDeleted` check in this method - I don't really see why that needs to happen in `run()`

Reply:
combining the isDelete check is one of my points - good to know that is possible